### ISSUE TITLE
Cannot read property 'addEventListener' of null - clipboard.min.js:7

### ISSUE THREAD

**Issue by hassaan-rana-** [zenorocha/clipboard.js#183]
I've linked the script but before using it I'm getting this error in console.

**Comment by kafeltz** [comment]
did you try wrapped in dom-ready event?

**Comment by hassaan-rana-** [comment]
I've just included this file and cannot use due to this jquery error.

**Comment by zenorocha** [comment]
Everything seems to be working over here (see [demo folder](https://github.com/zenorocha/clipboard.js/tree/master/demo)).

The only way we can help you is by providing a [minimal example](http://stackoverflow.com/help/mcve) on JSFiddle.

**Comment by cheekyboots** [comment]
I had the same problem and found the solution over here:
http://stackoverflow.com/questions/26107125/cannot-read-property-addeventlistener-of-null

Wrap the instantiation code in window.onload=function()

i.e. 

window.onload=function(){
    var clipboard = new Clipboard('.btn');
    clipboard.on('success', function(e) {
        console.log(e);
    });
    clipboard.on('error', function(e) {
        console.log(e);
    });
}

**Comment by michaelsomerville** [comment]
Having this issue, but it was working fine when the code was on the page. Moved the code it it's own js file and now it's not working.

**Comment by MikeWarren2014** [comment]
Does the dependency have to be **after** the element containing the data to be copied?! I'm having this problem as well. I put the dependency in the `<head>`, right after the jQuery dependency, and in my script file, I said 

`$(document).ready(function() { 
	// some other important stuff
	var clipboard = new Clipboard('#link');
});`
and it didn't work. However, when I moved that to the front of the script (before the `$(document).ready()` stuff) I got this error. MVCE coming in a few days....
